src/commands/list_folder.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-console */
+import { arweave } from 'ardrive-core-js';
+import { cliWalletDao, CLI_APP_NAME, CLI_APP_VERSION } from '..';
+import { ArDrive, ArDriveAnonymous } from '../ardrive';
+import {
+	ArFSDAO,
+	ArFSDAOAnonymous,
+	ArFSPrivateFileOrFolderWithPaths,
+	ArFSPublicFileOrFolderWithPaths
+} from '../arfsdao';
+import { CLICommand, ParametersHelper } from '../CLICommand';
+import { ArDriveCommunityOracle } from '../community/ardrive_community_oracle';
+import {
+	DrivePasswordParameter,
+	ParentFolderIdParameter,
+	SeedPhraseParameter,
+	WalletFileParameter
+} from '../parameter_declarations';
+
+function alphabeticalOrder(a: string, b: string) {
+	return a.localeCompare(b);
+}
+
+new CLICommand({
+	name: 'list-folder',
+	parameters: [ParentFolderIdParameter, SeedPhraseParameter, WalletFileParameter, DrivePasswordParameter],
+	async action(options) {
+		const parameters = new ParametersHelper(options);
+		const folderId = parameters.getRequiredParameterValue(ParentFolderIdParameter);
+		let children: (ArFSPrivateFileOrFolderWithPaths | ArFSPublicFileOrFolderWithPaths)[];
+
+		if (await parameters.getIsPrivate()) {
+			const wallet = await parameters.getRequiredWallet();
+			const arDrive = new ArDrive(
+				wallet,
+				cliWalletDao,
+				new ArFSDAO(wallet, arweave),
+				new ArDriveCommunityOracle(arweave),
+				CLI_APP_NAME,
+				CLI_APP_VERSION
+			);
+
+			const driveId = await arDrive.getDriveIdForFolderId(folderId);
+			const driveKey = await parameters.getDriveKey(driveId);
+
+			children = await arDrive.listPrivateFolder(folderId, driveKey);
+		} else {
+			const arDrive = new ArDriveAnonymous(new ArFSDAOAnonymous(arweave));
```

Comment:
Can we use the factory here?

Reply:
7f42bfed0a7cb350b8e1662f28b87f6161332d53